CHANGELOG.md

```diff
@@ -11,6 +11,9 @@
 - Fixed a bug where imports from permanent stage locations were ignored for temporary stored procedures, UDTFs, UDFs, and UDAFs.
 - Revert back to using CTAS (create table as select) statement for `Dataframe.writer.save_as_table` which does not need insert permission for writing tables.
 
+### New Features
+- Added support for Literal expression with VariantType `Column(Literal(10, VariantType()))`
```

Comment:
I don't think we need to mention this, `Literal` is not exposed to users

Reply:
I can remove it and give it the proper label. Should we maybe support then `to_variant(lit(10))` to create a properly typed literal?

Reply:
what do you mean "support then to_variant(lit(10))"? I meant just we don't need to anything in changelog since it's an internal class

Reply:
If we had something like to_variant(lit(10)), we could change the logic to

def to_variant(e: ColumnOrName) -> Column:
    """Converts any value to a VARIANT value or NULL (if input is NULL).

    Example::

        >>> df = session.create_dataframe([1, 2, 3, 4], schema=['a'])
        >>> df_conv = df.select(to_variant(col("a")).as_("ans"))
        >>> df_conv.collect()
        [Row(ANS='1'), Row(ANS='2'), Row(ANS='3'), Row(ANS='4')]

        After conversion via to_variant, another variant dataframe can be merged.

        >>> from snowflake.snowpark.types import VariantType, StructField, StructType
        >>> from snowflake.snowpark import Row
        >>> schema = StructType([StructField("a", VariantType())])
        >>> df_other = session.create_dataframe([Row(a=10), Row(a='test'), Row(a={'a': 10, 'b': 20}), Row(a=[1, 2, 3])], schema=schema)
        >>> df_conv.union(df_other).select(typeof(col("ans")).as_("ans")).collect()
        [Row(ANS='INTEGER'), Row(ANS='INTEGER'), Row(ANS='INTEGER'), Row(ANS='INTEGER'), Row(ANS='INTEGER'), Row(ANS='VARCHAR'), Row(ANS='OBJECT'), Row(ANS='ARRAY')]
    """
    
    if isinstance(c, Colunm) and isinstance(c._expression, Literal):
       return Column(Literal(c._expression.value, VariantType()))
    
    c = _to_col_if_str(e, "to_variant")
    return builtin("to_variant")(c)

Reply:
updated changelog.